tests/system/test_access.py

```diff
@@ -92,30 +92,36 @@ def setUp(self):
 
         user = os.getenv("ES_USER", "apm_server_user")
         password = os.getenv("ES_PASS", "changeme")
-        self.admin_es_url = self.get_elasticsearch_url(user, password)
+        self.es_url_apm_server_user = self.get_elasticsearch_url(user, password)
 
-        self.api_key_url = "{}/_security/api_key".format(self.admin_es_url)
+        self.api_key_url = "{}/_security/api_key".format(self.es_url_apm_server_user)
         self.api_key_name = "apm-systemtest"
+        self.privileges_url = "{}/_security/privilege".format(self.es_url_apm_server_user)
 
-        # clean setup: delete all existing api_keys of current user
+        # clean setup:
+        # delete all existing api_keys of current user
         requests.delete(self.api_key_url,
                         data=json.dumps({'name': self.api_key_name}),
                         headers=headers(content_type='application/json'))
         self.wait_until(lambda: self.api_keys_invalidated(), name="delete former api keys")
+        # delete all existing application privileges to ensure they can be created for current user
+        for privilege in ["sourcemap", "intake", "config"]:
+            requests.delete("{}/{}/{}".format(self.privileges_url, self.application, privilege))
+        self.wait_until(lambda: self.privileges_deleted(self.application))
```

Comment:
privileges_deleted now returns a tuple, so won't this return immediately?

Reply:
Ouch, yes that doesn't make sense.